ch_pipeline/analysis/timing.py

```diff
@@ -45,6 +51,36 @@ def setup(self, tcorr):
             Note that the timing correction must *already* be referenced with respect
             to the times that were used to calibrate if processing stacked data.
         """
+        # Query flag database for date ranges that need timing correction
+        needs_timing_correction_flags = None
+
+        if mpiutil.rank0:
```

Comment:
You should use `self.rank == 0` in here. Generally I'd try to stay away from `mpiutil` as much as you can, it's designed to help writing code in contexts where you might not have MPI installed, but I've found it's more confusion than it's worth. Code in `draco` you can basically assume MPI will be available if you need it, and so that functionality is baked into `SingleTask`.